src/main/java/com/work/management/utils/IndexConfigurator.java

```diff
@@ -0,0 +1,142 @@
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.client.indices.PutMappingRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+@Component
+final class IndexConfigurator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IndexConfigurator.class);
+
+  private final RestHighLevelClient client;
+  private final ElasticConfigProperties elasticConfigProperties;
+
+  @Autowired
+  public IndexConfigurator(RestHighLevelClient client,
+      ElasticConfigProperties elasticConfigProperties) {
+    this.client = client;
+    this.elasticConfigProperties = elasticConfigProperties;
+  }
+
+  @PostConstruct
+  private void createIndexMapping() {
+    try {
+
+      final GetIndexRequest request = new GetIndexRequest(
+          elasticConfigProperties.getIndex().getName());
+      final boolean exists = client.indices().exists(request, RequestOptions.DEFAULT);
+      if (!exists) {
+        final CreateIndexRequest indexRequest = new CreateIndexRequest(
+            elasticConfigProperties.getIndex().getName());
+        indexRequest.settings(Settings.builder()
+            .put("index.number_of_shards", elasticConfigProperties.getIndex().getShard())
+            .put("index.number_of_replicas", elasticConfigProperties.getIndex().getReplica()));
+        final CreateIndexResponse createIndexResponse = client.indices()
+            .create(indexRequest, RequestOptions.DEFAULT);
+        if (createIndexResponse.isAcknowledged() && createIndexResponse.isShardsAcknowledged()) {
+          LOGGER
+              .info("{} index created successfully", elasticConfigProperties.getIndex().getName());
+        } else {
+          LOGGER.debug("Failed to create {} index", elasticConfigProperties.getIndex().getName());
+        }
+
+        final PutMappingRequest mappingRequest = new PutMappingRequest(
+            elasticConfigProperties.getIndex().getName());
+
+        mappingRequest.source(createBuilderWithMapping());
+        final AcknowledgedResponse putMappingResponse = client.indices()
+            .putMapping(mappingRequest, RequestOptions.DEFAULT);
+
+        if (putMappingResponse.isAcknowledged()) {
+          LOGGER.info("Mapping of {} was successfully created",
+              elasticConfigProperties.getIndex().getName());
+        } else {
+          LOGGER
+              .debug("Creating mapping of {} failed", elasticConfigProperties.getIndex().getName());
+        }
+      }
+    } catch (Exception ex) {
+      LOGGER.error("An exception thrown in createIndexWithMapping method.", ex);
+    }
+  }
+
+  private XContentBuilder createBuilderWithMapping() {
+    try {
```

Comment:
Why is this needed?
And why are you creating same mapping for each and every call?
Can't you create it once and reuse it again and again?